mace/core/memory_optimizer.cc

```diff
@@ -187,7 +187,7 @@ void MemoryOptimizer::Optimize(
           // minimize add_mem_size; if best_mem_add_size is 0,
           // then minimize waste_mem_size
           if ((best_added_mem_size > 0 && added_mem_size < best_added_mem_size)
-              || (best_added_mem_size == 0 &&
+              || (best_added_mem_size == 0 && added_mem_size == 0 &&
```

Comment:
The `added_ mem_ Size 'is used to record the newly added memory size and cannot be used for conditional judgment. I    will close this pr,  please feel free to communicate if you have any questions.
anyway, Thanks for your contribution.